trace/ptrace/ptrace.go

```diff
@@ -438,6 +438,45 @@ func rangeActualScope(r exptrace.Range) rangeScope {
 	}
 }
 
+// runningGauge tracks maximum value of a gauge and creates a time series that can be rendered.
+type runningGauge struct {
+	// current is the current value of the gauge.
+	current int64
+	// currentTime is the time of the last update.
+	currentTime exptrace.Time
+	// series is a list of points that capture the current value.
+	series []Point
+}
+
+// add adds a value to the gauge at time t.
+// It assumes that t will not decrease with subsequent calls.
+func (rg *runningGauge) add(t exptrace.Time, v int64) {
+	if rg.currentTime == 0 {
+		rg.currentTime = t
+	}
+	switch {
+	case rg.currentTime == 0:
+		rg.currentTime = t
+	case t > rg.currentTime:
```

Comment:
Instead of dropping successive points for the same timestamp, we should record the last sample only.

Reply:
Previously the code was recording the last sample, that was achieved by lagging one behind — it only appended the last value.

Anyway, this is now replaced by the direct append.